### DISCUSSION TITLE
Free Software version of Outline, and rationales for upcoming licence change?

### DISCUSSION THREAD

**Discussion by almereyda** [outline/outline#3301] [Ideas]
I wanted to ask this for a while, so here we go to document the current state of affairs on the topic:

While it had been stated previously in multiple occassions, there is a claim that Outline will be released under a (permissive) open source licence which releases the source code as so-called free software. The current licence in use, BSL, is not considered a free software licence, and comes with certain restrictions for the developers providing code changes upstream here.

- https://github.com/outline/outline/blob/main/LICENSE

---

1. Is there still the plan to release Outline under a (permissive) Open Source licence?

- http://web.archive.org/web/20220326031152/http://www.gnu.org/licenses/license-list.html
- http://web.archive.org/web/20220326072546/https://opensource.org/licenses

2. In case not, which would be the reasons to secure business for Outline that prohibit it from being released as FLOSS (freely-licenced open source software)?

---

Thank you for keeping an eye on this.

**Comment by tommoor** [comment]
Hi @almereyda – the way the BSL license works is that the software moves to Apache License, Version 2.0 four years after it is initially released, this change is encoded in the license wording itself. For example, the latest version v0.62.0 will be available under the Apache license on the 1st of March 2026.

Sometimes the dates lag behind in the license text because it must be manually updated with every release, but the principal is the same.

**Reply by almereyda** [reply]
Thank you for detailing this out. This gives a prospect at when Outline turns into free software, or older versions already are ^_^.

It is not said that the codebase from a point release is still relevant four years thereafter, but this is not the concern here.

**Reply by tommoor** [reply]
Every version prior to `v0.40.0` was released under MIT, if you go that far back it's quite a different product though. No versions released under the BSL license have become OSS yet, the first will be `v0.40.2` on 2023-03-01

**Reply by almereyda** [reply]
That was yesterday 🎉

We could keep a log somewhere about which versions become free at what time.

Or look at the release log, and add three years to each publication date: https://github.com/outline/outline/releases?page=6

**Comment by binaryben** [comment]
Would the team consider moving to an open source license?

**Reply by tommoor** [reply]
Not really, the current license is a good balance that allows companies to self host while protecting our ability to monetize the project and keep it sustainable for the long term.

**Reply by binaryben** [reply]
To be fair, I like the sentiment of the licence. It just seems ambiguous about whether companies can self-host it without needing to purchase a licence or not. Can you please provide clarification or a link to more information on the licence? What I see online says you cannot use it for commercial purposes - but does that mean I can't setup Outline and on sell it or I cannot setup Outline for managing the docs of a new business venture?

**Reply by hendrikbl** [reply]
> but does that mean I can't setup Outline and on sell it or I cannot setup Outline for managing the docs of a new business venture?

Thats exactly what I'm trying to figure out right now. I'm looking for a simple and clean, selfhosted wiki alternative and outline looks promising.
But I have to be sure I can use it without purchasing a license.

**Reply by almereyda** [reply]
I fear that's not what the license says. When setting up a business, you will have access to legal counsel. I would suggest to ask them for interpreting the terms of the BSL, as I doubt it is the responsibility of the one to speak out a license having to do more than that.

**Reply by binaryben** [reply]
@almereyda We wouldn't have to seek legal counsel if less ambiguous licence was used or if the Outline team replied with a clear answer.

**Reply by tommoor** [reply]
As I mentioned above the intent of the license is to allow companies to self host the software for internal use.

**Reply by binaryben** [reply]
Thanks @tommoor, that will have to do in terms of clarification
